duktape/src/main/java/com/squareup/duktape/DuktapeException.java

```diff
@@ -15,8 +15,56 @@
  */
 package com.squareup.duktape;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public final class DuktapeException extends RuntimeException {
+  /**
+   * Duktape stack trace strings have multiple lines of the format "function filename.ext:line".
+   * "function" is optional, but we'll omit frames without a function, since it means the frame is
+   * in native code.
+   */
+  private final static Pattern STACK_TRACE_PATTERN =
+      Pattern.compile("\\s*([^\\s]+) ([^\\s]+):(\\d+).*$");
+  /** Java StackTraceElements require a class name.  We don't have one in JS, so use this. */
+  private final static String STACK_TRACE_CLASS_NAME = "JavaScript";
+
   public DuktapeException(String detailMessage) {
-    super(detailMessage);
+    super(getErrorMessage(detailMessage));
+    String[] lines = detailMessage.split("\n", -1);
+    if (lines.length <= 1) {
```

Comment:
(Actually, `split()` can return an empty array, but only in broken mode where you don't pass `-1` as the 2nd arg)